pkg/manifest/firstboot.go

```diff
@@ -0,0 +1,160 @@
+package manifest
+
+import (
+	"fmt"
+	"html/template"
+	"io/fs"
+	"strings"
+
+	"github.com/osbuild/images/internal/common"
+	"github.com/osbuild/images/pkg/customizations/firstboot"
+	"github.com/osbuild/images/pkg/customizations/fsnode"
+)
+
+// checkName prevents path traversal
```

Comment:
I think the comment is a bit misleading, the checkName function is much stricter, as a side effect it prevents path traversal but if that was the goal the name could be relaxed.